src/test/java/com/salesforce/mirus/metrics/MirrorJmxReporterTest.java

```diff
@@ -0,0 +1,49 @@
+package com.salesforce.mirus.metrics;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.metrics.Metrics;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class MirrorJmxReporterTest {
+
+  private MirrorJmxReporter mirrorJmxReporter;
+  private Metrics metrics;
+  private final String TEST_TOPIC = "TestTopic";
+
+  @Before
+  public void setUp() throws Exception {
+    mirrorJmxReporter = MirrorJmxReporter.getInstance();
+    metrics = mirrorJmxReporter.metrics;
+  }
+
+  @Test
+  public void updateLatencyMetrics() {
+    TopicPartition topicPartition = new TopicPartition(TEST_TOPIC, 1);
+    mirrorJmxReporter.addTopics(List.of(topicPartition));
+
+    mirrorJmxReporter.recordMirrorLatency(TEST_TOPIC, 500);
+    mirrorJmxReporter.recordMirrorLatency(TEST_TOPIC, TimeUnit.MINUTES.toMillis(5));
+    mirrorJmxReporter.recordMirrorLatency(TEST_TOPIC, TimeUnit.MINUTES.toMillis(10));
+
+    Map<String, String> tags = new LinkedHashMap<>();
+    tags.put("topic", TEST_TOPIC);
+    tags.put("bucket", "0m");
+    Object value =
+        metrics
+            .metrics()
+            .get(
+                metrics.metricName(
+                    MirrorJmxReporter.HISTOGRAM_LATENCY.name(),
+                    MirrorJmxReporter.HISTOGRAM_LATENCY.group(),
+                    MirrorJmxReporter.HISTOGRAM_LATENCY.description(),
+                    tags))
+            .metricValue();
+    Assert.assertNotNull(value);
```

Comment:
What do you think of also checking that there is exactly one occurrence per bucket?

Reply:
Regardless of whether we stick with the cumulative histogram implementation, shall we also check that the other buckets are empty?